### DISCUSSION TITLE
Cannot generate restraint.gro

### DISCUSSION THREAD

**Discussion by Toolxx** [Martini-Force-Field-Initiative/MartiniDiscussions#12] [Tutorials]
I am following the tutorial "https://cgmartini.nl/docs/tutorials/Martini3/ProteinsII/" and "https://cgmartini.nl/docs/tutorials/Martini3/ProteinsI/" and I found the issue that as I follow the each step, I did not generate "restraint.gro", and the "martinize2 -p" will generate nothing. It there someone meet the similar issue?

For example, I used "martinize2 -f 181L_clean.pdb -o t4l_only.top -x t4l_cg.pdb -dssp -p backbone"  It only generate "
 molecule_0.itp   t4l_cg.pdb            t4l_only.top"

How can I solve this issue? Is it from incompatible issue for python 3.8? Or some issue related to martinize2?

**Comment by csbrasnett** [comment]
Martinize2 generates position restraints within the `[ position_restraints ]` directive of the output itp file (unlike gmx2pdb, which writes them externally to be included).

In `gmx grompp ...` the `-r` restraint argument reads a coordinate file for position references to restrain to, using the parameters as specified in both the itp file in the `[ position_restraints ]` directive, and the `define = ...` of the mdp file.

For the purposes of the first tutorial, you can use the same position file for the initial structure and restraints, because `define = -DPOSRES -DFLEXIBLE` in the .mdp file only applies these to the soluble protein. You can then take a similar approach for the second tutorial too

**Comment by csbrasnett** [comment]
If anyone else is wondering about this same problem, I've expanded on my answer in the [vermouth discussion forum](https://github.com/marrink-lab/vermouth-martinize/discussions/618?converting=1)
